### PR TITLE
Update idaes-pse requirement in preparation to 2.1 release

### LINKED ISSUE #1049: Resolve compatibility issues with IDAES 2.1.0/Pyomo 6.6.1
As of today, several tests are still failing with IDAES 2.1.0rc0/Pyomo 6.6.1:

<details>
<summary>Screenshot (click to expand)</summary>

image

</details>

<details>
<summary>Log (click to expand)</summary>

```
=========================== short test summary info ============================
FAILED watertap/core/tests/test_zero_order_properties.py::test_CV_integration - idaes.core.util.exceptions.ConfigurationError: Property package has only one phase; control volume cannot include phase equilibrium terms. Some property packages support phase equilibrium implicitly in which case additional terms are not necessary.
FAILED watertap/examples/flowsheets/case_studies/activated_sludge/tests/test_asm2d_flowsheet.py::TestASM2DFlowsheet::test_results - assert 72813.23796132828 == 101325 ± 1.0e+01
  comparison failed
  Obtained: 72813.23796132828
  Expected: 101325 ± 1.0e+01
FAILED watertap/examples/flowsheets/case_studies/seawater_RO_desalination/tests/test_seawater_RO_desalination.py::test_seawater_RO_desalination_pressure_exchanger - pyomo.core.base.units_container.InconsistentUnitsError: Error in units found in expression: fs.ro_costing.maintenance_labor_chemical_operating_cost + fs.ro_costing.aggregate_fixed_operating_cost + fs.ro_costing.aggregate_variable_operating_cost + 0*fs.ro_costing.utilization_factor: USD_2018 / year not compatible with dimensionless.
FAILED watertap/examples/flowsheets/case_studies/seawater_RO_desalination/tests/test_seawater_RO_desalination.py::test_seawater_RO_desalination_pump_as_turbine - pyomo.core.base.units_container.InconsistentUnitsError: Error in units found in expression: fs.ro_costing.maintenance_labor_chemical_operating_cost + fs.ro_costing.aggregate_fixed_operating_cost + fs.ro_costing.aggregate_variable_operating_cost + 0*fs.ro_costing.utilization_factor: USD_2018 / year not compatible with dimensionless.
FAILED watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1575_hrcs/tests/test_multi_sweep.py::test_multi_sweep[1] - pyomo.core.base.units_container.InconsistentUnitsError: Error in units found in expression: fs.watertap_costing.maintenance_labor_chemical_operating_cost + fs.watertap_costing.aggregate_fixed_operating_cost + fs.watertap_costing.aggregate_variable_operating_cost + 0*fs.watertap_costing.utilization_factor: USD_2018 / year not compatible with dimensionless.
FAILED watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1575_hrcs/tests/test_multi_sweep.py::test_multi_sweep[2] - pyomo.core.base.units_container.InconsistentUnitsError: Error in units found in expression: fs.watertap_costing.maintenance_labor_chemical_operating_cost + fs.watertap_costing.aggregate_fixed_operating_cost + fs.watertap_costing.aggregate_variable_operating_cost + 0*fs.watertap_costing.utilization_factor: USD_2018 / year not compatible with dimensionless.
FAILED watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/tests/test_amo_1690.py::TestAMO1690Flowsheet::test_solve - ValueError: Cannot load a SolverResults object with bad status: error
FAILED watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/tests/test_amo_1690_sweep.py::test_sweep[1] - ValueError: Cannot load a SolverResults object with bad status: error
FAILED watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/tests/test_amo_1690_sweep.py::test_sweep[2] - ValueError: Cannot load a SolverResults object with bad status: error
FAILED watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/tests/test_amo_1690_sweep.py::test_sweep[3] - ValueError: Cannot load a SolverResults object with bad status: error
FAILED watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/swine_wwt/tests/test_swine_wwt.py::Test_Swine_WWT_Flowsheet::test_costing - pyomo.core.base.units_container.InconsistentUnitsError: Error in units found in expression: fs.watertap_costing.maintenance_labor_chemical_operating_cost + fs.watertap_costing.aggregate_fixed_operating_cost + fs.watertap_costing.aggregate_variable_operating_cost + 0*fs.watertap_costing.utilization_factor: USD_2018 / year not compatible with dimensionless.
FAILED watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/swine_wwt/tests/test_swine_wwt.py::Test_Swine_WWT_Flowsheet::test_display - ValueError: No value for uninitialized NumericValue object fs.costing.aggregate_variable_operating_cost
FAILED watertap/examples/flowsheets/lsrro/tests/test_lssro_multi_sweep.py::test_against_multisweep[1] - pyomo.core.base.units_container.InconsistentUnitsError: Error in units found in expression: (fs.costing.factor_capital_annualization*fs.PrimaryPumps[1].costing.capital_cost/(31557600.000000004*(s/a)*fs.product.properties[0.0].flow_vol_phase[Liq]*fs.costing.utilization_factor)) + 0*USD_2018 + (fs.costing.factor_capital_annualization*fs.EnergyRecoveryDevices[1].costing.capital_cost/(31557600.000000004*(s/a)*fs.product.properties[0.0].flow_vol_phase[Liq]*fs.costing.utilization_factor)): USD_2018 / meter ** 3 not compatible with USD_2018.
FAILED watertap/examples/flowsheets/lsrro/tests/test_lssro_paper_analysis.py::test_against_paper_analysis[10_nodes_A_5LMHbar_optimize_Bmax_nomax_125gL_35.0pct.csv-4] - pyomo.core.base.units_container.InconsistentUnitsError: Error in units found in expression: (fs.costing.factor_capital_annualization*fs.PrimaryPumps[1].costing.capital_cost/(31557600.000000004*(s/a)*fs.product.properties[0.0].flow_vol_phase[Liq]*fs.costing.utilization_factor)) + 0*USD_2018 + (fs.costing.factor_capital_annualization*fs.EnergyRecoveryDevices[1].costing.capital_cost/(31557600.000000004*(s/a)*fs.product.properties[0.0].flow_vol_phase[Liq]*fs.costing.utilization_factor)): USD_2018 / meter ** 3 not compatible with USD_2018.
FAILED watertap/property_models/tests/test_NDMA_prop_pack.py::TestNDMAroperty_idaes::test_default_initializer - NotImplementedError
FAILED watertap/property_models/tests/test_NaCl_prop_pack.py::TestNaClProperty_idaes::test_default_initializer - NotImplementedError
FAILED watertap/property_models/tests/test_cryst_prop_pack.py::TestNaClProperty_idaes::test_default_initializer - NotImplementedError
FAILED watertap/property_models/tests/test_seawater_prop_pack.py::TestSeawaterProperty_idaes::test_default_initializer - NotImplementedError
FAILED watertap/property_models/tests/test_water_prop_pack.py::TestSeawaterProperty_idaes::test_default_initializer - NotImplementedError
FAILED watertap/ui/tests/test_flowsheet_interfaces.py::test_roundtrip_with_garbage_collection[watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.amo_1690.amo_1690_ui-2 times] - RuntimeError: Building flowsheet: Cannot load a SolverResults object with bad status: error
FAILED watertap/ui/tests/test_flowsheet_interfaces.py::test_roundtrip_with_garbage_collection[watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.amo_1690.amo_1690_ui-3 times] - RuntimeError: Building flowsheet: Cannot load a SolverResults object with bad status: error
ERROR watertap/ui/tests/test_flowsheet_interfaces.py::TestFlowsheetInterface::test_build[watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.amo_1690.amo_1690_ui] - RuntimeError: Building flowsheet: Cannot load a SolverResults object with bad status: error
ERROR watertap/ui/tests/test_flowsheet_interfaces.py::TestFlowsheetInterface::test_model_objects[watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.amo_1690.amo_1690_ui] - RuntimeError: Building flowsheet: Cannot load a SolverResults object with bad status: error
ERROR watertap/ui/tests/test_flowsheet_interfaces.py::TestFlowsheetInterface::test_solve[watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.amo_1690.amo_1690_ui] - RuntimeError: Building flowsheet: Cannot load a SolverResults object with bad status: error
= 21 failed, 3090 passed, 3 skipped, 12 xfailed, 5 xpassed, 38 warnings, 3 errors in 1019.07s (0:16:59) =
```
</details>

This issue is to track progress made in resolving these.

---

- [x] `NotImplementedError` for `watertap/property_models/tests/test_cryst_prop_pack.py::test_default_initializer()`
- [x] Incompatible units
- [x] watertap/core/tests/test_zero_order_properties.py::test_CV_integration - idaes.core.util.exceptions.ConfigurationError: Property package has only one phase; control volume cannot include phase equilibrium terms. Some property packages support phase equilibrium implicitly in which case additional terms are not necessary.
- [x] @lbianchi-lbl Review dependencies issues (e.g. failures in `watertap/examples/chemistry/tests/test_notebooks.py`)

### PR DESCRIPTION
**Pull request by lbianchi-lbl**
## TODO

Leave this PR open until the following PRs have been merged, and check that those changes are compatible with IDAES 2.1/Pyomo 6.6

- [x] #1006 
- [x] #1054 
- [x] #1057 (not likely to be affected)
- [ ] #1062 (once merged)

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by andrewlee94** [comment]
The reason the tests are failing is because these specific tests are using the new `Initilaizer` objects in IDAES, but the property packages in use have not implemented the necessary `fix_initialization_states` methods.

This method needs to be implemented in the property packages, and needs to set the state blocks into a square form - generally fix the state variables and if necessary deactivate constraints that cannot be used when the states are fixed (e.g., sum of mole fractions). This is basically the same as the first step(s) of the older `initialize` methods. Note that this method does not need to return anything (unlike older implementations) as the `Initializer` is recording the initial state itself.

**Comment by agarciadiego** [comment]
@lbianchi-lbl I'm running this tests in IDAES 2.1 so some may fail if we are still using 2.0 in this branch. (haven't finished all the corrections)

File: setup.py [hunk 1]
```diff
@@ -21,14 +21,14 @@
...
-    "idaes-pse==2.0.*",  # from PyPI
+    "idaes-pse==2.1.*",  # from PyPI
 ]
 
 SPECIAL_DEPENDENCIES_FOR_PRERELEASE = [
     # update with a tag from the nawi-hub/idaes-pse
     # when a version of IDAES newer than the latest stable release from PyPI
     # will become needed for the watertap development
-    "idaes-pse==2.0.*",
+    "idaes-pse==2.0.0",
```
Comment (lbianchi-lbl) [inline setup.py]: @agarciadiego You can apply this change to use IDAES 2.1 in the CI runs:

```suggestion
    "idaes-pse==2.1.0rc0",
```

**Comment by agarciadiego** [comment]
@lbianchi-lbl all the test pass in Windows and Linux. I don't know where the `E   ModuleNotFoundError: No module named 'nbconvert'` is coming from

**Comment by lbianchi-lbl** [comment]
> @lbianchi-lbl all the test pass in Windows and Linux. I don't know where the `E ModuleNotFoundError: No module named 'nbconvert'` is coming from

@agarciadiego that's a dependency that used to be included with IDAES, but has been removed as part of IDAES/idaes-pse#1133. Adding it back as a testing dependency should resolve those failures.

File: setup.py [hunk 2]
```diff
@@ -80,7 +80,7 @@
         # primary requirements for unit and property models
         # maintainers: switch to SPECIAL_DEPENDENCIES_FOR_RELEASE when cutting a release of watertap
         *SPECIAL_DEPENDENCIES_FOR_PRERELEASE,
-        "pyomo>=6.2,<6.6",  # (also needed for units in electrolyte database (edb))
+        "pyomo>=6.6.1",  # (also needed for units in electrolyte database (edb))
```
Comment (adam-a-a) [inline setup.py:83]:
```suggestion
        "pyomo<=6.6.1",  # (also needed for units in electrolyte database (edb))
```
Any reason why this shouldn't be as suggested? Would tests fail using a version of pyomo below 6.6.1?

File: watertap/examples/flowsheets/case_studies/activated_sludge/ASM2D_flowsheet.py [hunk 3]
```diff
@@ -282,13 +282,13 @@ def scale_variables(m):
...
-                iscale.set_scaling_factor(var, 1e-3)
+                iscale.set_scaling_factor(var, 1e-4)
             if "enth_mol" in var.name:
                 iscale.set_scaling_factor(var, 1e-3)
             if "alkalinity" in var.name:
                 iscale.set_scaling_factor(var, 1e3)
-            if "conc_mass_comp" in var.name:
-                iscale.set_scaling_factor(var, 1e2)
+            # if "conc_mass_comp" in var.name:
+            #     iscale.set_scaling_factor(var, 1e2)
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/activated_sludge/ASM2D_flowsheet.py]: Should this be uncommented with the scaling factor set to 1 (based on tests which look to be passing across the board, except for experimental mac check)?

File: watertap/examples/flowsheets/case_studies/activated_sludge/tests/test_asm2d_flowsheet.py [hunk 4]
```diff
@@ -96,32 +96,32 @@ def test_results(self, model):
             0, abs=1e-4
         )
         assert value(model.fs.Treated.conc_mass_comp[0, "X_PAO"]) == pytest.approx(
-            2.21218e-3, abs=1e-4
+            2.111e-3, abs=1e-4
         )
         assert value(model.fs.Treated.conc_mass_comp[0, "X_PHA"]) == pytest.approx(
-            2.1218e-9, abs=1e-4
+            8.981e-5, abs=1e-4
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/activated_sludge/tests/test_asm2d_flowsheet.py:102]: `X_PHA` changed by 4 orders of magnitude but am unsure why.

File: watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py [hunk 5]
```diff
@@ -211,6 +211,32 @@ def build_flowsheet():
...
+        "S_A",
+        "X_I",
+        "X_H",
+        "X_PP",
+        "X_S",
+    ]
+
+    for i in skip_constraint:
+        m.fs.electroNP.solute_removal_equation[0.0, "Liq", i].deactivate()
+        m.fs.electroNP.solute_treated_equation[0.0, "Liq", i].deactivate()
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py:241]: WHy did these have to be deactivated? Was the flowsheet overconstrained somehow or were the default removal fractions of 0 causing problems?

Comment (agarciadiego) [reply watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py:241]: Yes, all of these constraint were over constraining the problem and degeneracy hunter found them as degenerate constraints

File: watertap/examples/flowsheets/case_studies/activated_sludge/ASM2D_flowsheet.py (hunk 3 above)
Comment (agarciadiego) [reply watertap/examples/flowsheets/case_studies/activated_sludge/ASM2D_flowsheet.py]: I think it should be deleted the `"conc_mass_comp"` are being scaled individually in the next lines

**Comment by agarciadiego** [comment]
@lbianchi-lbl @bknueven Do you have any objections to me skipping the mac tests for the tests that are failing?

**Comment by lbianchi-lbl** [comment]
> @lbianchi-lbl @bknueven Do you have any objections to me skipping the mac tests for the tests that are failing?

No objections from me. Skip away!

File: setup.py (hunk 2 above)
Comment (lbianchi-lbl) [reply setup.py:83]: Yes, Pyomo 6.6.0 has known issues that affect us, namely Pyomo/pyomo#2844 and IDAES/idaes-pse#1191. Both have been resolved in Pyomo 6.6.1.

File: watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py (hunk 5 above)
Comment (bknueven) [reply watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py:241]: Is this something that should be resolved in the unit model?

**Review by bknueven — APPROVED** [review]
Obviously there's a few things to investigate here, but I would be in favor of going ahead with this as-is.

**Comment by lbianchi-lbl** [comment]
It seems that merging in #1062 resulted in some test failures on Windows.

- pytest:
  image
- Jupyter notebook:
  image

Any thoughts on what could be done to try to address these? @adam-a-a @avdudchenko (based on the commits on #1062).

**Pull request merged**